### ISSUE TITLE
SvelteKit integration

### ISSUE THREAD

**Issue by 0gust1** [Azure/static-web-apps#390]
[SvelteKit](https://kit.svelte.dev) is heading toward the first release.

It uses `adapters` to deploy on various serverless environments: https://kit.svelte.dev/docs#adapters (already supported: Vercel, Begin, CloudFlare, Netlify). The productivity, dev experience and output quality is pretty impressive. 

It seems that it would be quite possible (and interesting) to also build for an "Azure static web apps" target (maybe the `adapter-node` would do the job).

I don't have the time right now to look into this (maybe in the future), but it may worth a closer look.

Be careful: the `adapter` part of SvelteKit is still moving (but the pace has decreased a lot).

**Comment by miwebst** [comment]
Thanks for sharing @0gust1! We also don't have a ton of time right now, but can look into this in the future. Seems cool from an initial glance :)

**Comment by aaronpowell** [comment]
@0gust1 SvelteKit is something that I'm interested in where it fits too.

I think that an adapter would have to be somewhere between `adapter-node` and `adapter-static`, as you need to generate a HTML page when the adapter outputs, as that's really what Static Web Apps will need, but you don't need to go full SSR (unless you want to).

It'd be worth digging into the way the server file is generated, because it does ultimately have the HTML that's needed, so it's a matter of learning what control there is over the way that's chunked for output.

**Comment by 0gust1** [comment]
Yes, and there are also routing configuration files to generate (I took a _really quick_ look this weekend to the netlify and vercel adapters to see how they do it). But atm, the adapter API is still obscure to me (I may have to ask in svelte discord to have some pointers/info on this).

**Comment by GeekTrainer** [comment]
[Hello world works fine](https://www.geektrainer.dev/code/deploy-sveltekit-aswa/) with `adapter-static`, but I'll openly admit I haven't ventured much beyond this. I'm happy to explore more and see what else I can discover.

**Comment by 0gust1** [comment]
An interesting example on how one could design and test a POC adapter for Azure static webapps (here, it's for AWS CDK):
https://dev.to/juranki/diy-sveltekit-cdk-adapter-3enp

**Comment by rob-balfre** [comment]
@geoffrich that's amazing work!  Thanks for sharing

**Comment by Fedeorlandau** [comment]
@geo

> I've created a [POC adapter](https://www.npmjs.com/package/svelte-adapter-azure-swa) and published it to NPM. It is definitely beta -- I've only tested it with the SvelteKit demo app -- but it seems to work fine so far. Here's the [deployed demo](https://delightful-bush-01bdb9a0f.azurestaticapps.net/). Note that the index and /about routes are prerendered to static files, while the /todos route is server rendered via an Azure function.
> 
> Conceptually, it's pretty simple. Similar to the [Netlify adapter](https://github.com/sveltejs/kit/tree/master/packages/adapter-netlify), it 1) builds the static files and 2) creates a serverless function that handles SSR and server-side endpoints. I set `navigationFallback` in the [SWA config](https://github.com/geoffrich/svelte-adapter-azure-swa/blob/main/files/staticwebapp.config.json) to call the Azure function if a path doesn't match (i.e., the HTML file for that route wasn't prerendered). This is what makes the SSR work. I had to define another catch-all route to intercept POST/PUT/DELETE requests as well so that you could POST to a server endpoint (navigationFallback seems to only allow GETs).
> 
> All the Azure function does is map the function request to the input the SvelteKit handler expects, call the handler, and return the output.
> 
> Feedback/issues appreciated! As I mentioned, this is very early and hasn't been tested that much, so I'm sure there's some issues.

Thank you for building this 👍

**Comment by geoffrich** [comment]
Update for anyone following this thread - v0.12 of svelte-adapter-azure-swa was [just released](https://github.com/geoffrich/svelte-adapter-azure-swa/releases/tag/v0.12.0) and it makes the process of releasing a SvelteKit app on Azure SWA almost zero-config. Here's the SvelteKit demo [running on Azure SWA](https://polite-desert-00b80111e.2.azurestaticapps.net/sverdle).

After initializing a new SvelteKit project with `npm create svelte@latest` and adding a SWA [build config file](https://learn.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions), you only need to set the correct [build locations](https://github.com/geoffrich/svelte-adapter-azure-swa#azure-configuration) and you should be able to deploy your app. 

Previous versions of the adapter required you to create a `api/render` folder and associated files yourself; with this release, that is no longer necessary.

Give it a try if you haven't already, and submit an issue on the repo if you run into any problems!
